files/image_config/warmboot-finalizer/finalize-warmboot.sh

```diff
@@ -0,0 +1,96 @@
+#! /bin/bash
+
+VERBOSE=no
+
+# Check components
+COMP_LIST="orchagent neighsyncd bgp"
+EXP_STATE="reconciled"
+
+ASSISTANT_SCRIPT="/usr/bin/neighbor_advertiser"
+
+
+function debug()
+{
+    /usr/bin/logger "WARMBOOT_FINALIZER : $1"
+    if [[ x"${VERBOSE}" == x"yes" ]]; then
+        echo `date` "- $1"
+    fi
+}
+
+
+function check_warm_boot()
+{
+    WARM_BOOT=`/usr/bin/redis-cli -n 6 hget "WARM_RESTART_ENABLE_TABLE|system" enable`
+}
+
+
+function wait_for_database_service()
+{
+    debug "Wait for database to become ready..."
+
+    # Wait for redis server start before database clean
+    until [[ $(/usr/bin/docker exec database redis-cli ping | grep -c PONG) -gt 0 ]];
+        do sleep 1;
+    done
+
+    # Wait for configDB initialization
+    until [[ $(/usr/bin/docker exec database redis-cli -n 4 GET "CONFIG_DB_INITIALIZED") ]];
+        do sleep 1;
+    done
+
+    debug "Database is ready..."
+}
+
+
+function get_component_state()
+{
+    /usr/bin/redis-cli -n 6 hget "WARM_RESTART_TABLE|$1" state
+}
+
+
+function check_list()
+{
+    RET_LIST=''
+    for comp in $@; do
+        state=`get_component_state ${comp}`
+	if [[ x"${state}" != x"${EXP_STATE}" ]]; then
+            RET_LIST="${RET_LIST} ${comp}"
+	fi
+    done
+
+    echo ${RET_LIST}
+}
+
+
+function finalize_warm_boot()
+{
+    debug "Finalizing warmboot..."
+    sudo config warm_restart disable
```

Comment:
don't need sudo here

Reply:
Last login: Mon Apr  1 15:22:27 2019 from 10.20.48.209
admin@str-7260cx3-acs-2:~$ config warm_restart enable 
Root privileges are required for this operation
admin@str-7260cx3-acs-2:~$ config warm_restart disable 
Root privileges are required for this operation

Reply:
@yxieca but the service runs as root?

Reply:
Good point. I guess extra sudo doesn't hurt? :-)